### DISCUSSION TITLE
Get rid of groovy tests?

### DISCUSSION THREAD

**Discussion by sebastian-peter** [ie3-institute/simona#645] [General]
Since there's only been a handful of groovy tests, I think it might make sense just getting rid of them.

Arguments pro:
- Less maintenance effort (there's been some hours poured into this):
  - CI
  - code formatting
  - dependency updates
  - ...
- Less languages/test frameworks to learn for beginners
- In some parts, referencing Scala code in groovy is cumbersome
- Tests could be converted to ScalaTest, which also provides data driven testing

Arguments contra:
- Some, but not too much effort to convert
- Data driven testing in ScalaTest is slightly less neat/convenient

Let's discuss!

**Comment by t-ober** [comment]
I'm totally for getting rid of the groovy tests.

**Reply by danielfeismann** [reply]
Same here. I think these are good tasks for hiwi's

**Comment by sebastian-peter** [comment]
Ok I'd say we have an agreement here, I opened #646
